src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -0,0 +1,431 @@
+        datasetFieldMappings.add(
+            new DatasetFieldMapping(dsVersion.getUuid(), datasetFieldRow.getUuid()));
+      }
+    }
+    datasetFieldDao.updateFieldMapping(datasetFieldMappings);
+
+    if (isInput) {
+      runDao.updateInputMapping(runUuid, dsVersion.getUuid());
+    }
+
+    jobVersionDao.upsertDatasetIoMapping(
+        jobVersion.getUuid(), dataset.getUuid(), isInput ? IoType.INPUT : IoType.OUTPUT);
+  }
+
+  default SourceType getSourceType(LineageDataset ds) {
+    return SourceType.POSTGRESQL;
+  }
+
+  default DatasetType getDatasetType(LineageDataset ds) {
+    return DatasetType.DB_TABLE;
+  }
+
+  default RunState getRunState(String eventType) {
+    switch (eventType.toLowerCase()) {
+      case "complete":
+        return RunState.COMPLETED;
+      case "abort":
+        return RunState.ABORTED;
+      case "fail":
+        return RunState.FAILED;
+      case "start":
+        return RunState.RUNNING;
+      default:
+        return RunState.NEW;
+    }
+  }
+
+  default Map<String, String> createRunArgs(LineageEvent event) {
+    Map<String, String> args = new LinkedHashMap<>();
+    if (event.run.facets != null) {
+      if (event.run.facets.nominalTime != null) {
+        args.put(
+            "run.facets.nominalTime.nominalStartTime",
+            event.run.facets.nominalTime.nominalStartTime.toString());
+        if (event.run.facets.nominalTime.nominalEndTime != null) {
+          args.put(
+              "run.facets.nominalTime.nominalEndTime",
+              event.run.facets.nominalTime.nominalEndTime.toString());
+        }
+      }
+      if (event.run.facets.parent != null) {
+        args.put("run.facets.parent.run.runId", event.run.facets.parent.run.runId);
+        args.put("run.facets.parent.job.name", event.run.facets.parent.job.name);
+        args.put("run.facets.parent.job.namespace", event.run.facets.parent.job.namespace);
+      }
+    }
+    return args;
+  }
+
+  default UUID buildJobVersion(LineageEvent event, Map<String, String> context) {
+    final byte[] bytes =
+        VERSION_JOINER
+            .join(event.job.namespace, event.job.name, event.producer, KV_JOINER.join(context))
+            .getBytes(UTF_8);
+    return UUID.nameUUIDFromBytes(bytes);
+  }
+
+  default Map<String, String> buildJobContext(LineageEvent event) {
+    Map<String, String> args = new LinkedHashMap<>();
+    if (event.job.facets != null) {
+      if (event.job.facets.sourceCodeLocation != null) {
+        if (event.job.facets.sourceCodeLocation.type != null) {
+          args.put("job.facets.sourceCodeLocation.type", event.job.facets.sourceCodeLocation.type);
+        }
+        if (event.job.facets.sourceCodeLocation.url != null) {
+          args.put("job.facets.sourceCodeLocation.url", event.job.facets.sourceCodeLocation.url);
+        }
+      }
+      if (event.job.facets.sql != null) {
+        args.put("job.facets.sql.query", event.job.facets.sql.query);
```

Comment:
should this just be "sql"

Reply:
What about the other facets for the job context?

Reply:
I think we should add a job_facets column in the JobVersion or possibly a JobFacets table linked to the JobVersion.

Reply:
Might make sense to add a `job_facets` table as facets may not change from version to version (similar in approach to how we handle _job.contexts_)